fdbserver/storageserver.actor.cpp

```diff
@@ -1454,10 +1452,15 @@ ACTOR Future<Void> getKeyValues( StorageServer* data, GetKeyValuesRequest req )
 				data->metrics.notify(r.data[i].key, m);
 			}*/
 
+			// For performance concerns, the cost of a range read is billed to the start key and end key of the range.
+			int64_t totalByteSize = 0;
 			for (int i = 0; i < r.data.size(); i++) {
-				StorageMetrics m;
-				m.bytesReadPerKSecond = std::max((int64_t)r.data[i].expectedSize(), SERVER_KNOBS->EMPTY_READ_PENALTY);
-				data->metrics.notify(r.data[i].key, m);
+				totalByteSize += r.data[i].expectedSize();
+			}
+			if (totalByteSize > 0 && SERVER_KNOBS->READ_SAMPLING_ENABLED) {
```

Comment:
I think the branch can be faster if `SERVER_KNOBS->READ_SAMPLING_ENABLED`  is moved ahead of `totalByteSize > 0`, especially when `READ_SAMPLING_ENABLED  = false`.

Reply:
It won't be a big deal since `READ_SAMPLING_ENABLED` will be true in most cases.

Reply:
I understand that. But for the case who don't enable this flag, it will pay the small penalty.